Microsoft.Toolkit.Uwp.UI/Cache/CacheBase.cs

```diff
@@ -79,17 +89,84 @@ public int MaxMemoryCacheCount
         }
 
         /// <summary>
+        /// Gets instance of <see cref="HttpClient"/>
+        /// </summary>
+        protected HttpClient HttpClient
+        {
+            get
+            {
+                if (_httpClient != null)
+                {
+                    return _httpClient;
+                }
+
+                var messageHandler = new HttpClientHandler() { MaxConnectionsPerServer = 10 };
+
+                _httpClient = new HttpClient(messageHandler);
+
+                return _httpClient;
+            }
+        }
+
+        /// <summary>
         /// Initializes FileCache and provides root folder and cache folder name
         /// </summary>
         /// <param name="folder">Folder that is used as root for cache</param>
+        /// <returns>awaitable task</returns>
+        public Task InitializeAsync(StorageFolder folder)
+        {
+            return InitializeAsync(folder, null, null);
+        }
+
+        /// <summary>
+        /// Initializes FileCache and provides root folder and cache folder name
+        /// </summary>
         /// <param name="folderName">Cache folder name</param>
         /// <returns>awaitable task</returns>
-        public virtual async Task InitializeAsync(StorageFolder folder, string folderName)
+        public Task InitializeAsync(string folderName)
+        {
+            return InitializeAsync(null, folderName, null);
+        }
+
+        /// <summary>
+        /// Initializes FileCache and provides root folder and cache folder name
+        /// </summary>
+        /// <param name="httpMessageHandler">instance of <see cref="HttpMessageHandler"/></param>
+        /// <returns>awaitable task</returns>
+        public Task InitializeAsync(HttpMessageHandler httpMessageHandler)
+        {
+            return InitializeAsync(null, null, httpMessageHandler);
+        }
+
+        /// <summary>
+        /// Initializes FileCache and provides root folder and cache folder name
+        /// </summary>
+        /// <param name="folder">Folder that is used as root for cache</param>
+        /// <param name="folderName">Cache folder name</param>
+        /// <returns>awaitable task</returns>
+        public Task InitializeAsync(StorageFolder folder, string folderName)
+        {
+            return InitializeAsync(folder, folderName, null);
+        }
+
+        /// <summary>
+        /// Initializes FileCache and provides root folder and cache folder name
+        /// </summary>
+        /// <param name="folder">Folder that is used as root for cache</param>
+        /// <param name="folderName">Cache folder name</param>
+        /// <param name="httpMessageHandler">instance of <see cref="HttpMessageHandler"/></param>
+        /// <returns>awaitable task</returns>
+        public virtual async Task InitializeAsync(StorageFolder folder, string folderName, HttpMessageHandler httpMessageHandler)
```

Comment:
I personally hate having to set things to null when calling methods.. I am not fussed.. I can reverted to original calls only